### ISSUE TITLE
Make this repo a fork of the api

### ISSUE THREAD

**Issue by ZachOhara** [Thunderbots-5604/2015-Code_OLD#2]
@danielgrimshaw @pranavmathur @danielgrimshaw @zachw000 @danielgrimshaw @danielgrimshaw @danielgrimshaw 

One of the goals of the API is to make the code as hardware-generic as possible, so that the same code could ideally be run on any NXT+Tetrix system that uses RobotC. The point in making the API its own repository was to make the code more readily available to anyone who might want to use it, at the disadvantage of having to make changes in different locations depending on the type of change. 

My proposed solution to this is that we keep the API as its own repository, on the Thunderbots' page, and make our 2015-Code a fork of this repository. That way, all changes relevant to the 2015 robot design can be easily made in a single place, and we can keep the API repository publicly visible. This has the added advantage of merging and pull requests between the repositories, as well as being able to fetch from upstream if the 2015 branch of the api is modified directly.

@pranavmathur has already agreed to this
@danielgrimshaw please vote
@zachw000 please vote

**Comment by danielgrimshaw** [comment]
I don't recall @pranavmathur agreeing to keep the Robot API repo on the Github. If I remember correctly, he was the first one to request its deletion. We can re add the API as a repo as soon as it is fully completed. Also, I don't think that you can fork your own code. Therefore my vote is no for now, but it will change to yes as soon as we get gyro working as well as full PID control calibrated and working.

**Comment by ZachOhara** [comment]
Pranav just agreed over the phone. The API will never be fully completed, until the day it simply stops being maintained. Currently, it is in a working state, and anything that would be in the master branch would be in a working state. The 2015 branch of the API repo would be the source for the 2015 repo.

The contents of the master branch of Robot-API are currently the exact same form as the api of 2014-code, so we needn't worry about it not working. As more features are added to the 2015 api, they can all be merged into the api repo as soon as they are tested, which currently includes the PID to some extent.

Basically, we can release a working version now and not worry about it, and then make it work better as the development of 2015-code continues.

**Comment by danielgrimshaw** [comment]
Fine then, but lets release the working version after we clean up the
code a little and implement the correct PID constants. I hereby
officially change my vote to yes.

**Comment by danielgrimshaw** [comment]
@pranavmathur has told me that he never agreed to this. @ZachOhara revert all changes based off of this assumption. Also, my vote was no until we get all of the pid constants set.

**Comment by pranavmathur** [comment]
One issue I think we need to address is how long we want to work on the api. @ZachOhara says that he wants it to be platform generic and work with any robot, but that's not our goal. Our goal is to have a place to store our code; We have no customer. I think we should have a 2015-code repo with 2-3 branches, but no api or old code repos. The api is not so important that it needs a repository. For the 2015-2016 season, we should finish working on the api very early (e.g. after we decide on a drive system) and then only add to it when adding functions for new hardware or completely full functions (e.g. not working on a timer system for a month). We should focus our time and effort on top-level programs, and work on the api when functionality is needed for a top-level program.

@danielgrimshaw

**Comment by ZachOhara** [comment]
Our goal is to write effective code for our hardware.  The problem with
this is that none of us can see the future, so we don't know what hardware
we have or won't have. The only perfect way around this is to write code
that works on any hardware we may end up using in the future. I recognize
the near-impossibility of that, but we should definitely try to hit as
broad a range of hardware as possible.

I agree with you that we should refine the API before the next season and
focus efforts on top-level programs, but the thing with that is that
sometimes it's just better to add functionality to the API instead of
directly to the top-level.
